### ISSUE TITLE
can tsd server support metrics aggregated by background thread in advance?

### ISSUE THREAD

**Issue by juliennju** [OpenTSDB/opentsdb#524]
we have a lot of metrics data in opentsdb and wait for a long time for quering them, I think we can aggregate them to fewer metrics in advance, as we usally know how to aggregate metrics data before quering.

**Comment by juliennju** [comment]
I mean pre-aggregation when metrics with cardinality a million tagkv. Is there any plan or thougt to solve it?

**Comment by manolama** [comment]
Yeah, we actually have an implementation to upstream that will let you store aggregated/rolled up data processed out of band.

**Comment by juliennju** [comment]
@manolama , Would you please share the solution, we are urgent for it. And we can also help to solve it.

**Comment by manolama** [comment]
@juliennju Upstreaming the storage and query code for rollups now in https://github.com/OpenTSDB/opentsdb/tree/put/src/rollup. TSDs won't be able to perform aggregations in a background thread as it's too intensive at scale to read the data (e.g. problem we had with compactions). Instead we need an external system that performs the rollups/aggs. We're trying to get Yahoo's Storm based code open sourced that handles this.
